mailbox/store/src/test/java/org/apache/james/mailbox/store/mail/model/AttachmentTest.java

```diff
@@ -42,11 +42,81 @@ public void streamShouldBeConsumedOneTime() throws Exception {
     @Test
     public void streamShouldBeConsumedMoreThanOneTime() throws Exception {
         String input = "mystream";
-        Attachment attachment = new Attachment(input.getBytes());
+        Attachment attachment = Attachment.from(input.getBytes(), "content");
 
         attachment.getStream();
         InputStream stream = attachment.getStream();
         assertThat(stream).isNotNull();
         assertThat(IOUtils.toString(stream)).isEqualTo(input);
     }
+
+    @Test (expected = IllegalArgumentException.class)
+    public void builderShouldThrowWhenAttachmentIdIsNull() {
+        Attachment.builder()
+            .attachmentId(null);
+    }
+
+    @Test (expected = IllegalArgumentException.class)
+    public void builderShouldThrowWhenBytesIsNull() {
+        Attachment.builder()
+            .bytes(null);
+    }
+
+    @Test (expected = IllegalArgumentException.class)
+    public void builderShouldThrowWhenTypeIsNull() {
+        Attachment.builder()
+            .type(null);
+    }
+
+    @Test (expected = IllegalArgumentException.class)
+    public void builderShouldThrowWhenTypeIsEmpty() {
+        Attachment.builder()
+            .type("");
+    }
+
+    @Test (expected = IllegalStateException.class)
+    public void buildShouldThrowWhenAttachmentIdIsNotProvided() {
+        Attachment.builder().build();
+    }
+
+    @Test (expected = IllegalStateException.class)
+    public void buildShouldThrowWhenBytesIsNotProvided() {
+        Attachment.builder()
+            .attachmentId(AttachmentId.forPayload("mystream".getBytes()))
+            .build();
+    }
+
+    @Test (expected = IllegalStateException.class)
+    public void buildShouldThrowWhenTypeIsNotProvided() {
+        Attachment.builder()
+            .attachmentId(AttachmentId.forPayload("mystream".getBytes()))
+            .bytes("mystream".getBytes())
+            .build();
+    }
+
+    @Test (expected = IllegalStateException.class)
+    public void buildShouldThrowWhenSizeIsNotProvided() {
```

Comment:
I introduce two static methods, which are responsible of doing such process.
The builder has only one responsibility: build the Attachment.